### PR TITLE
Document AllowedFlexVolumes in PSP

### PR DESCRIPTION
**Pull request by tengqm**
Closes: #6447

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6967)

### CODE REVIEW

File: docs/concepts/policy/pod-security-policy.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ administrator to control the following:
 | Usage of host networking and ports                  | [`hostNetwork`, `hostPorts`](#host-namespaces) |
 | Usage of volume types                               | [`volumes`](#volumes-and-file-systems)      |
 | Usage of the host filesystem                        | [`allowedHostPaths`](#volumes-and-file-systems) |
+| Usage of FlexVolume drivers                         | [`allowedFlexVolumes](#flexvolume-drivers) |
```
Comment (heckj) [inline docs/concepts/policy/pod-security-policy.md]: missing a backtick at the end of `allowedFelxVolumes` here

File: docs/concepts/policy/pod-security-policy.md [hunk 2]
```diff
@@ -417,6 +418,20 @@ containers, and abusing the credentials of system services, such as Kubelet._
...
 
+### FlexVolume drivers
+
+When the [`Volumes`](#volumes-and-file-systems) field contains `flexVolume` in
+its list value, the cluster admin can further specify which driver(s) is permitted
+by setting the `allowedFlexVolumes` field.
+
+**AllowedFlexVolumes** - Provides a whitelist of allowed FlexVolumes. Empty or
+nil indicates that all FlexVolume drivers may be used. For example, the following
+setting permits the `examle/fast_cache` driver to be used on nodes:
```
Comment (heckj) [inline docs/concepts/policy/pod-security-policy.md]: is "permits the" correct, or is "only permits the" more accurate?

Comment (tengqm) [reply docs/concepts/policy/pod-security-policy.md]: you are right.

File: docs/concepts/policy/pod-security-policy.md (hunk 1 above)
Comment (tengqm) [reply docs/concepts/policy/pod-security-policy.md]: ah ... will fix.

**Comment by tengqm** [comment]
@heckj fixed.

**Comment by php-coder** [comment]
I have couple questions:
- why this were written from scratch instead of backporting/adapting https://github.com/kubernetes/website/pull/6563 from @wanghaoran1988 ?
- why this PR wasn't reviewed by (at least) one of the approvers from the list in the header of the document (@pweil- and @tallclair)?

File: docs/concepts/policy/pod-security-policy.md [hunk 3]
```diff
@@ -417,6 +418,20 @@ containers, and abusing the credentials of system services, such as Kubelet._
...
+When the [`Volumes`](#volumes-and-file-systems) field contains `flexVolume` in
+its list value, the cluster admin can further specify which driver(s) is permitted
+by setting the `allowedFlexVolumes` field.
+
+**AllowedFlexVolumes** - Provides a whitelist of allowed FlexVolumes. Empty or
+nil indicates that all FlexVolume drivers may be used. For example, the following
+setting only permits the `examle/fast_cache` driver to be used on nodes:
+
+```yaml
+allowedFlexVolumes: [ "example/fast_cache" ]
```
Comment (php-coder) [inline docs/concepts/policy/pod-security-policy.md]: Unfortunately, this example incorrect and won't work :-|

`allowedFlexVolumes` accepts list of the objects (not list of the strings). See correct example in the original PR from @wanghaoran1988

Comment (tengqm) [reply docs/concepts/policy/pod-security-policy.md]: My fault. It should be a list of objects where each object has a driver property. Will fix.

**Pull request merged**